web/web.go

```diff
@@ -0,0 +1,183 @@
+package web
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/transactions/db"
+	"github.com/transactions/entity"
+	"github.com/transactions/uow"
+	"gorm.io/gorm"
+	"net/http"
+)
+
+func New(subDB db.SubDB, todoDB db.TodoDB, todoUOW uow.TodoUnitOfWork) *mux.Router {
+	router := mux.NewRouter().StrictSlash(true)
+
+	c := newController(todoUOW, todoDB, subDB)
+	router.HandleFunc("/todos", c.CreateTodo).Methods(http.MethodPost)
+	router.HandleFunc("/todos/{todo_id}", c.UpdateTodo).Methods(http.MethodPut)
+	router.HandleFunc("/todos/{todo_id}", c.GetTodo).Methods(http.MethodGet)
+	router.HandleFunc("/subs/{sub_id}", c.UpdateSub).Methods(http.MethodPut)
+
+	return router
+}
+
+type ctrl struct {
+	todoUOW uow.TodoUnitOfWork
+	todoDB  db.TodoDB
+	subDB   db.SubDB
+}
+
+func newController(todoUOW uow.TodoUnitOfWork, todoDB db.TodoDB, subDB db.SubDB) *ctrl {
+	return &ctrl{todoUOW: todoUOW, todoDB: todoDB, subDB: subDB}
+}
+
+func (c *ctrl) CreateTodo(w http.ResponseWriter, r *http.Request) {
+	var todo entity.Todo
+	err := json.NewDecoder(r.Body).Decode(&todo)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	todo.ID = uuid.New()
+	if len(todo.Subs) > 0 {
+		for i := range todo.Subs {
+			todo.Subs[i].ID = uuid.New()
+			todo.Subs[i].TodoID = todo.ID
+		}
+	}
+
+	err = c.todoUOW.CreateTodo(todo)
```

Comment:
for me this **todoUOW** looks like a renaming of repository. What is diff between **uow** and **repo** in your implementation?